**Actividad 1 — Identificación de la fuente**

Preguntas
1.	**¿Quién produjo originalmente los datos?**                                                                        
    Los datos fueron producidos originalmente por la Organización de las Naciones Unidas (ONU), específicamente mediante World Population Prospects (2024).
2.	**¿Qué organización los procesa para su utilización en Our World in Data?** <br>
    Los datos son procesados por Our World in Data (OWID) para su utilización en su plataforma.
3.	**¿Qué representa la variable que analizarás?** <br>
    La variable que se analizará representa el cambio anual de la población, es decir, la variación que presenta la población de una entidad de un año a otro. El conjunto de datos se identifica como “Annual change in population

**Actividad 2 — Carga y exploración de datos**

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("../data/raw/annual-population-growth/annual-population-growth.csv")
#df: dataframe = es una tabla donde en el eje y hay etiquetas y en el eje x hay indices pero tambien pueden ser etiquetas

In [47]:
df.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [4]:
df.shape

(38400, 5)

In [5]:
df.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [6]:
df.dtypes

Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [7]:
df.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [8]:
df.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

In [9]:
df["Entity"].nunique()

256

Preguntas:
1.	**¿Cuántas filas y columnas tiene el dataset?**<br>
    Filas: 38400 <br>
    Columnas: 5
2.	**¿Qué tipo de datos contiene la variable Year?**<br>
    La variable Year es de tipo int64, es decir, contiene números enteros
3.	**¿Existen valores faltantes?**<br>
    Sí, el conjunto de datos contiene valores faltantes
4.	**¿Cuántas entidades diferentes contiene el dataset?**<br>
    Contiene 256 entidades diferentes.

**Actividad 3 — Estadística descriptiva**

In [10]:
anio_consultado = 2023

In [75]:
df_anio_consultado = df[df["Year"]== anio_consultado]

In [12]:
df_anio_consultado.shape

(256, 5)

In [85]:
# Seleccionar la variable de cambio anual de población
variable = df_anio_consultado["Annual change in population"].dropna()

print("Media:", variable.mean())
print("Mediana:", variable.median())
print("Desviación estándar:", variable.std())
print("Mínimo:", variable.min()) #df_anio_consultado.sort_values("Annual change in population").head(1)
print("Máximo:", variable.max()) #df_anio_consultado.sort_values("Annual change in population").tail(1)

Media: 2008097.765625
Mediana: 46865.5
Desviación estándar: 9078962.549083743
Mínimo: -3315935.0
Máximo: 72315107.0


Preguntas:
1.	**¿Cuál es el valor de la media?** <br>
    La media es de 2,008,097.765625 personas
2.	**¿Cuál es el valor de la mediana?** <br>
    La mediana es de 46,865.5 personas
3.	**¿Cuál es el valor de la desviación estándar?** <br>
    La desviación estándar es de 9,078,962.549083743 personas
4.	**¿Qué entidad presenta el valor mínimo y cuál el máximo?** <br>
    El mínimo corresponde a Ukraine (-3,315,935) y el máximo a Less developed regions, excluding China (72,315,107)

**Actividad 4 — Comprobación del entorno**

In [14]:
import sys

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)


Python: 3.14.7 (tags/v3.14.7:823f032, Aug  5 2026, 10:51:32) [MSC v.1944 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2


**Trabajo en clase**                             <br> 17/08/2026

In [15]:
#Serie [] 1 corchete
df_year = df["Year"] 
df_year.head()

0    1951
1    1952
2    1953
3    1954
4    1955
Name: Year, dtype: int64

In [16]:
#DataFrame [[]] 2 corchetes, al poner dos corchetes se hace un dataframe porque esto le esta indicando que tiene dos dimensiones
df_year = df[ ["Year"] ]
df_year.head()

,Year
0,1951
1,1952
2,1953
3,1954
4,1955


In [17]:
#Extracción por columnas
df_extraccion_columnas = df[["Entity","Year"]]
df_extraccion_columnas.head()

,Entity,Year
0,Afghanistan,1951
1,Afghanistan,1952
2,Afghanistan,1953
3,Afghanistan,1954
4,Afghanistan,1955


In [18]:
#Extracción por filas --> Filas que cumplen una condición
#Estamos filtrando datos especificos 
# Para poder filtrar/extraer necesito primero preparar mi dataframe --> df[ ]
#Cada filtro se pone entre parentesis 

df_extraccion_filas = df[df["Year"]== 2020]
df_extraccion_filas.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
69,Afghanistan,AFG,2020,1212852.0,NaN
219,Africa (UN),UN_AFR,2020,32815675.0,NaN
369,Albania,ALB,2020,-13061.0,NaN
519,Algeria,DZA,2020,747543.0,NaN
669,American Samoa,ASM,2020,-451.0,NaN


In [19]:
#Filtrar dos entidades con un valor especifico 
#Cada filtro/condicion se pone entre parentesis 

df_2020_mex = df[ (df["Year"]== 2020) & (df["Code"]== "MEX") ]
df_2020_mex.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
22269,Mexico,MEX,2020,1036075.0,NaN


In [20]:
#Filtrar tres entidades con un valor especifico --> estamos usando tres condiciones 
#Cada filtro/ condicion  se pone entre parentesis 

df_2006_2020_mex = df[(df["Year"]>= 2006) & (df["Year"] <= 2020) &(df["Code"]== "MEX")]
df_2006_2020_mex.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
22255,Mexico,MEX,2006,1442162.0,NaN
22256,Mexico,MEX,2007,1520690.0,NaN
22257,Mexico,MEX,2008,1599930.0,NaN
22258,Mexico,MEX,2009,1625434.0,NaN
22259,Mexico,MEX,2010,1624178.0,NaN


In [21]:
#Filtrar tres entidades pero ahora solo con tres columnas 

df_extraccion_rango_mex_EYA = df_2006_2020_mex[["Entity","Year","Annual change in population"]]
df_extraccion_rango_mex_EYA.head()


,Entity,Year,Annual change in population
22255,Mexico,2006,1442162.0
22256,Mexico,2007,1520690.0
22257,Mexico,2008,1599930.0
22258,Mexico,2009,1625434.0
22259,Mexico,2010,1624178.0


In [22]:
df_extraccion_rango_mex_EYA.mean(numeric_only=True)

Year                           2.013000e+03
Annual change in population    1.399170e+06
dtype: float64

**Trabajo en clase** <br>
19/08/2026

Para extraer datos: dos corchetes y adentro el nombre de la columna que quiero extraer --> df_new = df [[nombre de la columna]]

df.loc[filas,columnas] <-- etiquetas <br>
df.iloc[filas,columnas] <-- posición (la posición simpre es numerico)

Ejemplo <br>
* df.iloc[10]  Regresa la fila que ocupa la posición 10 <br>
* df.iloc[:,3] Regresa todas las filas de la columna que ocupa la posición 3 <br> <br>

* df.loc[10]  Regresa la fila cuya etiqueta de indice es igual a 10

In [26]:
df.loc[:,["Entity","Year"]]

,Entity,Year
0,Afghanistan,1951
1,Afghanistan,1952
2,Afghanistan,1953
3,Afghanistan,1954
4,Afghanistan,1955
...,...,...
38395,Zimbabwe,2096
38396,Zimbabwe,2097
38397,Zimbabwe,2098
38398,Zimbabwe,2099


In [30]:
df.iloc[30:35,0:12]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
30,Afghanistan,AFG,1981,-1231727.0,NaN
31,Afghanistan,AFG,1982,-946206.0,NaN
32,Afghanistan,AFG,1983,-73395.0,NaN
33,Afghanistan,AFG,1984,272237.0,NaN
34,Afghanistan,AFG,1985,236633.0,NaN


*Ejercicios*
1. loc - Filas: Selecciona la fila cuyo indice sea 10000 

In [31]:
df.loc[10000]

Entity                                  Equatorial Guinea
Code                                                  GNQ
Year                                                 2051
Annual change in population                           NaN
Annual population change (Projected)              46407.0
Name: 10000, dtype: object

2. loc - Columnas: Selecciona todas las filas y unicamente las columnas Entity, Year y Annual change in population

In [37]:
df.loc[:,["Entity","Year","Annual change in population"]]
#al ser 3 datos los que queremos filtrar van entre parentesis (en esta caso son los del parentesis morado) esto es poque esos datos juntos generan una lista

,Entity,Year,Annual change in population
0,Afghanistan,1951,103163.0
1,Afghanistan,1952,108441.0
2,Afghanistan,1953,108919.0
3,Afghanistan,1954,111251.0
4,Afghanistan,1955,119027.0
...,...,...,...
38395,Zimbabwe,2096,NaN
38396,Zimbabwe,2097,NaN
38397,Zimbabwe,2098,NaN
38398,Zimbabwe,2099,NaN


3. Combina ambos

In [36]:
df.loc[10000,["Entity","Year","Annual change in population"]]
#los dos puntos no porque ya tengo un indice definido, es decir que ya tengo un indice de filas y quiero el arreglo de columnas

Entity                         Equatorial Guinea
Year                                        2051
Annual change in population                  NaN
Name: 10000, dtype: object

4. loc - Filtro: Selecciona todos los registros de Mexico correspondientes al año 2020

In [39]:
df.loc[(df["Entity"]== 'Mexico') & (df["Year"] == 2020)]


,Entity,Code,Year,Annual change in population,Annual population change (Projected)
22269,Mexico,MEX,2020,1036075.0,NaN


5. iloc - Filas: Selecciona las primeras 5 filas del DataFrame

In [87]:
df.iloc[0:5]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


6. iloc - Columnas: Selecciona todas las filas de las columnas que ocupan las posiciones 0,2 y 3

In [90]:
df.iloc[["0","2","3"]]

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN


7. iloc - Filas y columnas: Selecciona las filas de la posición 0 a la 4 y las columnas de la posición 0 a la 3

In [88]:
df.iloc[0:5, 0:4]

,Entity,Code,Year,Annual change in population
0,Afghanistan,AFG,1951,103163.0
1,Afghanistan,AFG,1952,108441.0
2,Afghanistan,AFG,1953,108919.0
3,Afghanistan,AFG,1954,111251.0
4,Afghanistan,AFG,1955,119027.0


Actividad 3 <br>
--> Año 2006 - 2023<br>
--> País que comience con la letra de su primer apellido

In [82]:
dff = df[(df ["Year"] >=2006) & (df ["Year"] <=2023) & (df["Entity"]=="Cuba")]

In [83]:
df_extraccion = dff [["Annual change in population"]]
df_extraccion.head()

,Annual change in population
8005,15997.0
8006,8724.0
8007,4490.0
8008,2955.0
8009,2695.0


In [84]:
print("Media:", df_extraccion.mean())
print("Mediana:", df_extraccion.median())
print("Desviación estándar:", df_extraccion.std())
print("Mínimo:", df_extraccion.min()) 
print("Máximo:", df_extraccion.max()) 

Media: Annual change in population   -13478.944444
dtype: float64
Mediana: Annual change in population   -10082.5
dtype: float64
Desviación estándar: Annual change in population    21625.973916
dtype: float64
Mínimo: Annual change in population   -62347.0
dtype: float64
Máximo: Annual change in population    15997.0
dtype: float64


Preguntas:
1.	**¿Cuál es el valor de la media?** -13478.944444 <br>
2.	**¿Cuál es el valor de la mediana?** -10082.5 <br>
3.	**¿Cuál es el valor de la desviación estándar?** 21625.973916 <br>
4.	**¿Qué entidad presenta el valor mínimo y cuál el máximo?** <br>
    El mínimo corresponde a Cuba -62347.0 y el máximo a Cuba con 15997.0

Actividad 4

In [77]:
import sys

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)

Python: 3.14.7 (tags/v3.14.7:823f032, Aug  5 2026, 10:51:32) [MSC v.1944 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2
